### DISCUSSION TITLE
在不需要完整istio的情况下，如何集成aeraki 到higress

### DISCUSSION THREAD

**Discussion by someview** [higress-group/higress#376] [Q&A]
在不需要完整istio的情况下，如何集成aeraki或者 meta-protocol-envoy 到higress中呢，能够支持私有的协议流量的治理。
由于higress和aeraki的流程原理，依赖关系不了解，不知道如何如何下手，寻求帮助

**Comment by johnlanni** [comment]
我对aeraki没有很多了解，看看社区其他同学有了解的吗 cc @Xunzhuo

**Comment by someview** [comment]
经过探索，当前并没有良好的对接私有协议的方式，除了aeraki-mesh下的metaprotocol proxy和阿里的mosn以外，未查看到公开的案例.
私有协议对接云原生网关，对接服务网格，没有现成的api 规范。metaprotocol proxy和mosn都和服务网格绑定了。认为这更应该是envoy对外提供的功能，提供类似于将私有协议转换成osi模型中的七层协议的功能，然后可以复用现有的7层负载均衡。
当前的envoy直接扩展来添加对私有协议的支持，难度还是比较大的，劝退一大波人。
